modules/dnn/src/model.cpp

```diff
@@ -1364,6 +1364,12 @@ struct TextDetectionModel_DB_Impl : public TextDetectionModel_Impl
         CV_Assert(outs.size() == 1);
         Mat binary = outs[0];
 
+        //Support ppdetect
+        int dim=binary.dims;
+        if(dim==3){
```

Comment:
I checked the models and it looks like the fix is reasonable. It makes sense to squash dimensions only if the extra dim has size 1. I propose to add `CV_assert` to raise issue, of some other model with unexpected output layout is used.

Reply:
I think we should edit the model graphs in [this](https://github.com/opencv/opencv_zoo/pull/66) PR, so that they have the right output shape, since in opencv model zoo models are tuned to work with OpenCV. This change is odd on its own.